### PR TITLE
Doc 7968 collection management

### PR DESCRIPTION
**Pull request by osfameron**

### CODE REVIEW

**Comment by osfameron** [comment]
NB: first 4 commits are as per https://github.com/couchbase/docs-sdk-nodejs/pull/106 still pending OK from @brett19

**Review by RichardSmedley — COMMENTED** [review]
LGTM,
as with https://github.com/couchbase/docs-sdk-nodejs/pull/106/
will merge after @brett19 code review.

**Review by brett19 — APPROVED** [review]
lgtm :)

**Pull request merged**
